test/docfx.Test/lib/MonikerRangeParserTest.cs

```diff
@@ -112,34 +114,55 @@ public MonikerRangeParserTest()
             "azure-sqldw-latest azuresqldb-current")]
         public void TestEvaluateMonikerRange(string rangeString, string expectedMonikers)
         {
-            var result = _monikerRangeParser.Parse(new SourceInfo<string>(rangeString)).ToList();
+            var (_, monikers) = _monikerRangeParser.Parse(new SourceInfo<string>(rangeString));
+            var result = monikers.ToList();
             result.Sort(_monikerComparer);
             Assert.Equal(expectedMonikers, string.Join(' ', result));
         }
 
         [Theory]
-        [InlineData("netcore-xp", "Moniker `netcore-xp` is not defined")]
-        [InlineData("netcore-1.0 < || netcore-2.0", "Expect a moniker string, but got ` || netcore-2.0`")]
-        [InlineData(">netcore&-1.0", "Parse ends before reaching end of string, unrecognized string: `&-1.0`")]
-        [InlineData(">=>netcore&-1.0", "Expect a moniker string, but got `>netcore&-1.0`")]
-        [InlineData(">netcore<-1.0", "Moniker `netcore` is not defined")]
-        [InlineData(">netcore<-1.0 ||| >netcore-2.0", "Expect a comparator set, but got `| >netcore-2.0`")]
-        [InlineData(">netcore<-1.0 || ||", "Expect a comparator set, but got ` ||`")]
-        [InlineData(">netcore<-1.0 || <", "Expect a moniker string, but got ``")]
+        [InlineData("netcore-xp", "Invalid moniker range 'netcore-xp': Moniker 'netcore-xp' is not defined")]
+        [InlineData("netcore-1.0 < || netcore-2.0", "Expect a moniker string, but got ' || netcore-2.0'")]
+        [InlineData(">netcore&-1.0", "Parse ends before reaching end of string, unrecognized string: '&-1.0'")]
+        [InlineData(">=>netcore&-1.0", "Expect a moniker string, but got '>netcore&-1.0'")]
+        [InlineData(">netcore<-1.0", "Invalid moniker range '>netcore<-1.0': Moniker 'netcore' is not defined")]
+        [InlineData(">netcore<-1.0 ||| >netcore-2.0", "Expect a comparator set, but got '| >netcore-2.0'")]
+        [InlineData(">netcore<-1.0 || ||", "Expect a comparator set, but got ' ||'")]
+        [InlineData(">netcore<-1.0 || <", "Expect a moniker string, but got ''")]
         public void InvalidMonikerRange(string rangeString, string errorMessage)
         {
-            var exception = Assert.Throws<DocfxException>(() => _monikerRangeParser.Parse(new SourceInfo<string>(rangeString)));
-            Assert.Equal("moniker-range-invalid", exception.Error.Code);
-            Assert.Equal(errorMessage, exception.Error.Message.Substring($"Invalid moniker range: '{rangeString}': ".Length));
+            var (errors,_) = _monikerRangeParser.Parse(new SourceInfo<string>(rangeString));
+            Assert.Contains(Errors.Versioning.MonikerRangeInvalid(null, errorMessage), errors, new ErrorEqualityComparer());
```

Comment:
This `ErrorEqualityComparer` seems odd
```suggestion
            Assert.Contains(errorMessage, errors.Select(error => error.Message));
```

Reply:
sounds good